core/adapters/eth_tx.go

```diff
@@ -191,6 +191,11 @@ func addReceiptToResult(
 		}
 	}
 
+	if receipt == nil {
```

Comment:
This should probably go at line 185 right? Save some work...

Reply:
@j16r Steve suggested this, but I kind of like having the nil check right next to the dereference.

Reply:
If you're adding a nil receipt to a result, this seems like a programming error, not a runtime error.